packages/drivers/driver-base/src/documentDeltaConnection.ts

```diff
@@ -419,7 +419,7 @@ export class DocumentDeltaConnection
 		);
 	}
 
-	protected disconnect(err: IAnyDriverError) {
+	protected disconnect = (err: IAnyDriverError) => {
```

Comment:
@jason-ha related to the discussion [here](https://github.com/microsoft/FluidFramework/pull/23550#discussion_r1976022216). If we need to do this, it feels like something that should go in its own PR, no? I don't see how it's relevant to this version of the change @RishhiB is making.

Reply:
@RishhiB if we apply this I think we also need to do it for `disconnectCore`, which is the one this PR is overriding. Still recommend getting Jason's eyes on this before we merge it.